src/ra_deleters.h

```diff
@@ -0,0 +1,211 @@
+#ifndef RA_DELETERS_H
+#define RA_DELETERS_H
+
+// specialization of std::default_delete doesn't work anymore
+// No need to specify the deleter as it's already taken care of
+
+namespace ra {
+/* Users shouldn't use these directly as it'll seem more complicated than it is, use the type aliases instead */
+namespace detail {
+
+// Handles have to types the way they are or restrict won't work
+// We can remove the const in front if setting to nullptr is desired
+
+struct CFileDeleter { void operator()(_In_ FILE* const __restrict fp) const noexcept { std::fclose(fp); } };
+struct MenuDeleter { void operator()(_In_ HMENU__* const __restrict hMenu) const noexcept { ::DestroyMenu(hMenu); } };
+
+// can also be used for HMODULE
+struct InstanceDeleter
+{
+    void operator()(_In_ HINSTANCE__* const  __restrict hInst) const noexcept
```

Comment:
To make a true "in" param, the const has to be in front. The object is deleted but not the pointer. If we want to null them out we can get rid of the const.

If I didn't use their struct names, it won't compile. 
`HINSTANCE __restrict hInst; -> resolves to: restrict HINSTANCE. `which is wrong
The way it is resolves correctly: `HINSTANCE__* const  restrict hInst`